src/libYARP_os/src/yarp/os/impl/TcpCarrier.h

```diff
@@ -19,7 +19,7 @@ namespace impl {
 /**
  * Communicating between two ports via TCP.
  */
-class TcpCarrier :
+class YARP_os_impl_API TcpCarrier :
```

Comment:
This is the most critical part of this carrier, and that's the reason why I want to keep this experimental for now. In order to be able to derive from `yarp::os::impl::TcpCarrier` we are forced to export the carrier in the ABI...
Anyway, let's keep it this way for now, we can think about something later...